### DISCUSSION TITLE
exploring-the-dotnet-8-preview-rendering-blazor-components-to-a-string/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#400] [Announcements]
# exploring-the-dotnet-8-preview-rendering-blazor-components-to-a-string/

In this post I describe how to render Blazor components to a string, how to use HtmlRenderer in ASP.NET Core, and show what does and doesn't work

https://andrewlock.net/exploring-the-dotnet-8-preview-rendering-blazor-components-to-a-string/

**Comment by fleed** [comment]
Interesting article (and series)!
Would it be possible to use `HtmlRender` to generate email content out of templates outside of a web application?
I would like to use the component in an Azure function to render (and send) emails

**Reply by benh57** [reply]
Also looking to use this for emails. It would be cool to see an example of taking a .razor file and generating it into a string.

**Comment by mikaelwallen** [comment]
So we've are using this approach and started rendering a few separate simple Razor components to HTML. However we ran into issues trying to use the CSS isolation feature. The case is as follows and is quite simple. Component A uses css isolation and is used in component B. When rendering HTML code for component B the css for component A is not included in the generated code. Is this a limitation or can it be done somehow?

**Reply by AlexeiPoplutin** [reply]
Same issue here, did you find a resolution?

**Comment by geertdoornbos** [comment]
Andrew, the SSR abbreviation in context of Blazor rendering means Server Side Rendering, so, the static rendermode is [Static Server Side Rendering](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-modes?view=aspnetcore-8.0). Keep up your good work!

**Comment by jupagose** [comment]
Great Job thanks!!

**Comment by atechdude** [comment]
Andrew, thanks for the article sir. I spent some time on this and came up with the following.

https://github.com/atechdude/RenderBlazorComponentToString/tree/master

**Comment by seankearon** [comment]
Lovely stuff - thank you!  I came here after following the article from the docs that you reference.  I just couldn't get layouts to work following the docs.  I was using `@layout MyLayout ` instead of `<LayoutView Layout="@typeof(MyLayout)"> Stuff here </LayoutView>`.

Is there a reason why  `@layout MyLayout ` doesn't work?

**Comment by daveylad** [comment]
Hi, landed here when researching how to generate email content using the ```HtmlRenderer```.

Passing parameters to the component to be rendered is fairly trivial, however I also need to pass some data to the layout that my components use (via ```LayoutView```)

Does anyone have any advice on the best way to achieve this?
